### PR TITLE
Add auth0_tenant data source

### PR DESCRIPTION
**Pull request by impl**
## Description

This is a replacement for #84, because I didn't realize changing the branch name would close that PR. :disappointed: 

It addresses the feedback in https://github.com/auth0/terraform-provider-auth0/pull/84#pullrequestreview-916011642, including an added example. The original description is below for posterity:

This change adds a new data source, `auth0_tenant`, which simply returns some helpful information about the current provider configuration. It is particularly helpful when developing a reusable module that expects this provider as an input, because the module may need to know the domain for downstream configuration (e.g. of frontend applications for redirects) or to set up client grants against the management API.

## Checklist

**Note:** Checklist required to be completed before a PR is considered to be reviewable.

#### Auth0 Code of Conduct
- [x] I have read and agreed to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

#### Auth0 General Contribution Guidelines
- [x] I have read the [Auth0 General Contribution Guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)

#### Changes include test coverage?
- [x] Yes
- [ ] Not needed

#### Does the description provide the correct amount of context?
- [x] Yes, the description provides enough context for the reviewer to understand what these changes accomplish

#### Have you updated the documentation?
- [x] Yes, I've updated the appropriate docs
- [ ] Not needed

#### Is this code ready for production?
- [x] Yes, all code changes are intentional and no debugging calls are left over

### CODE REVIEW

**Review by willvedd — DISMISSED** [review]
Great work and thanks for the contribution!

File: auth0/data_source_auth0_tenant.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	api := m.(*management.Management)
+
+	u, err := url.Parse(api.URI())
+	if err != nil {
+		return fmt.Errorf("unable to determine management API URL: %w", err)
+	}
+
+	d.SetId(resource.UniqueId())
+	d.Set("domain", u.Hostname())
+	d.Set("management_api_identifier", u.String())
```
Comment (sergiughf) [inline auth0/data_source_auth0_tenant.go:38]: Let's make sure we don't ignore these errors. We can make use of the multierror pkg in this case. e.g.

```
result := multierror.Append(d.Set("domain", u.Hostname() ....)

return result.ErrorOrNil()

File: auth0/data_source_auth0_tenant_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	"os"
+	"testing"
+
+	"github.com/auth0/terraform-provider-auth0/auth0/internal/random"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+const testAccDataTenantConfig = `
+
```
Comment (sergiughf) [inline auth0/data_source_auth0_tenant_test.go:14]:
```suggestion
```

**Review by sergiughf — APPROVED** [review]
This is an excellent contribution @impl Thank you honestly so much for this and all the hard work put into it. 💪🏻 

Apologies it took a while to get to this point. We're merging this in and will make a release till the end of the week most likely.

**Pull request merged**